src/transformers/integrations/integration_utils.py

```diff
@@ -1125,6 +1126,16 @@ def on_train_end(self, args, state, control, **kwargs):
                 self._experiment.clean()
                 self._initialized = False
 
+    def on_predict(self, args, state, control, metrics, **kwargs):
+        if not self._initialized:
+            self.setup(args, state, model=None)
+        if state.is_world_process_zero:
+            if self._experiment is not None:
+                rewritten_metrics = rewrite_logs(metrics)
+                self._experiment.__internal_api__log_metrics__(
+                    rewritten_metrics, step=state.global_step, epoch=state.epoch, framework="transformers"
+                )
```

Comment:
nit 

```suggestion
        if state.is_world_process_zero and if self._experiment is not None:
            rewritten_metrics = rewrite_logs(metrics)
            self._experiment.__internal_api__log_metrics__(
                rewritten_metrics, step=state.global_step, epoch=state.epoch, framework="transformers"
            )
```

Reply:
Fixed in the latest commit.
There is an extra `if` in your suggestion :)